cpu/atmega_common/periph/gpio.c

```diff
@@ -195,8 +281,70 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
         return -1;
     }
 
-    /* not a valid interrupt pin */
+    /* not a valid interrupt pin. Set as pcint instead if pcints are enabled */
     if (int_num < 0) {
+	    /* If pin change interrupts are enabled, enable mask and interrupt */
+        #ifdef PCINT_NUM_BANKS
+        int8_t offset = -1;
+        //uint8_t port_num = _port_num(pin);
+        uint8_t pin_num = _pin_num(pin);
+        uint8_t b;
+        for(b=0; b<PCINT_NUM_BANKS; b++) {
+            for(uint8_t p=0; p<8; p++) {
+                if(pin == pcint_mapping[b][p]) {
+                    offset = b*8+p;
+                    //pcint_lookup[ _port_num(pin) * 8 + pin_num ] = i;
+                    break;
+                }
+            }
+        }
+        /* if pcint was not found: return -1  */
+        if(offset < 0) {
+            return offset;
+        }
+
+        /* save configuration for pin change interrupt */
+        pcint_config[offset].pin     = pin;
+        pcint_config[offset].flank   = flank;
+        pcint_config[offset].arg     = arg;
+        pcint_config[offset].cb      = cb;
+
+        /* init gpio */        
+        gpio_init(pin, mode);
+        /* configure pcint */
+        cli();
+        switch (b) {
```

Comment:
Shouldn't the cases below be placed in `#ifdef MODULE_PCINT<NUM>` ... `#endif`?

Reply:
absolutely. And was also buggy until now!